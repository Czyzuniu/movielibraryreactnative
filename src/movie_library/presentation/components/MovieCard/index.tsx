import {Box, Pressable} from 'native-base';
import React from 'react';
import SimplisticMovie from '../../../domain/entity/SimplisticMovie';
import MoviePosterImage from '../MoviePosterImage';

type Props = {
  movie: SimplisticMovie;
  onPress: () => void;
};

function MovieCard({movie, onPress}: Props) {
  return (
    <Pressable onPress={onPress} testID={'MOVIE_CARD_' + movie.id}>
      {({isPressed}) => {
        return (
          <Box
            margin={1.5}
            opacity={isPressed ? 0.5 : 1}
            style={{
              transform: [
                {
                  scale: isPressed ? 0.96 : 1,
                },
              ],
            }}>
            <MoviePosterImage
              imagePath={movie.poster}
              resizeMode={'cover'}
              width={120}
              height={200}
            />
          </Box>
        );
      }}
    </Pressable>
  );
}

export default React.memo(MovieCard);
