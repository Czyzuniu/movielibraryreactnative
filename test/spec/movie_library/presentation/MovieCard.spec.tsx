import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import ContextWrapper from '../../../utils/ContextWrapper';
import TestObjects from '../../../utils/TestObjects';
import MovieCard from '../../../../src/movie_library/presentation/components/MovieCard';

describe('MovieCard Tests', () => {
  it('should render and match snapshot', () => {
    const {toJSON} = render(
      <ContextWrapper>
        <MovieCard movie={TestObjects.SimplisticMovie()} onPress={jest.fn} />
      </ContextWrapper>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should call on press prop when tapped', () => {
    const onPress = jest.fn();
    const movie = TestObjects.SimplisticMovie();
    const {getByTestId} = render(
      <ContextWrapper>
        <MovieCard movie={movie} onPress={onPress} />
      </ContextWrapper>,
    );

    fireEvent.press(getByTestId('MOVIE_CARD_' + movie.id));

    expect(onPress).toBeCalled();
  });
});
