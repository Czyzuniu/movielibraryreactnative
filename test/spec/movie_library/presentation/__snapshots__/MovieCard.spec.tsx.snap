// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard Tests should render and match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    dataSet={Object {}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={Object {}}
    testID="MOVIE_CARD_12345"
  >
    <View
      dataSet={Object {}}
      style={
        Array [
          Object {
            "flex": 1,
            "marginBottom": 6,
            "marginLeft": 6,
            "marginRight": 6,
            "marginTop": 6,
            "opacity": 1,
          },
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          },
        ]
      }
    >
      <Image
        accessibilityLabel="movie-poster"
        alt="movie-poster"
        dataSet={Object {}}
        onError={[Function]}
        path="poster.jpg"
        resizeMode="cover"
        source={
          Object {
            "uri": "https://mock-image-api.co.uk/w500/poster.jpg",
          }
        }
        style={
          Object {
            "borderRadius": 3,
            "height": 200,
            "maxWidth": "100%",
            "width": 120,
          }
        }
        testID="TEST_IMAGE_poster.jpg"
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
